fission/src/mirabuf/MirabufLoader.ts

```diff
@@ -118,24 +118,40 @@ class MirabufCachingService {
      * @returns {Promise<MirabufCacheInfo | undefined>} Promise with the result of the promise. Metadata on the mirabuf file if successful, undefined if not.
      */
     public static async CacheRemote(fetchLocation: string, miraType?: MiraType): Promise<MirabufCacheInfo | undefined> {
-        if (miraType) {
+        if (miraType !== undefined) {
             const map = MirabufCachingService.GetCacheMap(miraType)
             const target = map[fetchLocation]
+            if (target) return target
+        }
+        try {
+            // grab file remote
+            const miraBuff = await fetch(
+                encodeURI(fetchLocation),
+                import.meta.env.DEV ? { cache: "no-store" } : undefined
+            )
+                .then(x => x.blob())
+                .then(x => x.arrayBuffer())
```

Comment:
Right now you assume `fetch(...)` always resolves to a 2xx. If the server returns 404 or 500, you’ll end up turning that error into a blob/arrayBuffer error later. We should throw in a quick:

```ts
if (!resp.ok) throw new Error(...)
```

Before we do that just to make sure we clearly fail on a bad response.